plugin/pkg/scheduler/extender_test.go

```diff
@@ -0,0 +1,302 @@
+	}
+	return false, nil
+}
+
+func machine2PredicateExtender(pod *api.Pod, node *api.Node) (bool, error) {
+	if node.Name == "machine2" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func errorPrioritizerExtender(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error) {
+	return &schedulerapi.HostPriorityList{}, fmt.Errorf("Some error")
+}
+
+func machine1PrioritizerExtender(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error) {
+	result := schedulerapi.HostPriorityList{}
+	for _, node := range nodes.Items {
+		score := 1
+		if node.Name == "machine1" {
+			score = 10
+		}
+		result = append(result, schedulerapi.HostPriority{node.Name, score})
+	}
+	return &result, nil
+}
+
+func machine2PrioritizerExtender(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error) {
+	result := schedulerapi.HostPriorityList{}
+	for _, node := range nodes.Items {
+		score := 1
+		if node.Name == "machine2" {
+			score = 10
+		}
+		result = append(result, schedulerapi.HostPriority{node.Name, score})
+	}
+	return &result, nil
+}
+
+func machine2Prioritizer(_ *api.Pod, podLister algorithm.PodLister, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
+	nodes, err := nodeLister.List()
+	if err != nil {
+		return []schedulerapi.HostPriority{}, err
+	}
+
+	result := []schedulerapi.HostPriority{}
+	for _, node := range nodes.Items {
+		score := 1
+		if node.Name == "machine2" {
+			score = 10
+		}
+		result = append(result, schedulerapi.HostPriority{node.Name, score})
+	}
+	return result, nil
+}
+
+type FakeExtender struct {
+	predicates   []fitPredicate
+	prioritizers []priorityConfig
+	weight       int
+}
+
+func (f *FakeExtender) Filter(pod *api.Pod, nodes *api.NodeList) (*api.NodeList, error) {
+	filtered := []api.Node{}
+	for _, node := range nodes.Items {
+		fits := true
+		for _, predicate := range f.predicates {
+			fit, err := predicate(pod, &node)
+			if err != nil {
+				return &api.NodeList{}, err
+			}
+			if !fit {
+				fits = false
+				break
+			}
+		}
+		if fits {
+			filtered = append(filtered, node)
+		}
+	}
+	return &api.NodeList{Items: filtered}, nil
+}
+
+func (f *FakeExtender) Prioritize(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, int, error) {
+	result := schedulerapi.HostPriorityList{}
+	combinedScores := map[string]int{}
+	for _, prioritizer := range f.prioritizers {
+		weight := prioritizer.weight
+		if weight == 0 {
+			continue
+		}
+		priorityFunc := prioritizer.function
+		prioritizedList, err := priorityFunc(pod, nodes)
+		if err != nil {
+			return &schedulerapi.HostPriorityList{}, 0, err
+		}
+		for _, hostEntry := range *prioritizedList {
+			combinedScores[hostEntry.Host] += hostEntry.Score * weight
+		}
+	}
+	for host, score := range combinedScores {
+		result = append(result, schedulerapi.HostPriority{Host: host, Score: score})
+	}
+	return &result, f.weight, nil
+}
+
+func TestGenericSchedulerWithExtenders(t *testing.T) {
```

Comment:
It's best practice in Go to put testing functions on top and other helper functions (e.g. errorPredicateExtender, etc.) in the bottom of the file.